src/coll/mod.rs

```diff
@@ -788,6 +789,54 @@ impl<T> Collection<T> {
             .await?;
         Ok(())
     }
+
+    /// Starts a new [`ChangeStream`](change_stream/struct.ChangeStream.html) that receives events
+    /// for all changes in this collection. A
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html) cannot be started on system
+    /// collections.
+    ///
+    /// See the documentation [here](https://docs.mongodb.com/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read concern. Anything else
+    /// will cause a server error.
+    #[allow(unused)]
+    pub(crate) async fn watch(
+        &self,
+        options: impl Into<Option<ChangeStreamOptions>>,
+    ) -> Result<ChangeStream<ChangeStreamEvent<T>>>
+    where
+        T: DeserializeOwned + Unpin + Send + Sync,
+    {
+        todo!()
+    }
+
+    /// Starts a new [`ChangeStream`](change_stream/struct.ChangeStream.html) that receives events
+    /// for all changes in this collection and processes them via the given aggregation pipeline. A
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html) cannot be started on system
+    /// collections.
+    ///
+    /// See the documentation [here](https://docs.mongodb.com/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read concern. Anything else
+    /// will cause a server error.
+    ///
+    /// Also note that using a `$project` stage to remove any of the `_id`, `operationType` or `ns`
+    /// fields will cause an error. The driver requires these fields to support resumability. For
+    /// more information on resumability, see the documentation for
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html)
+    #[allow(unused)]
+    pub(crate) async fn watch_with_pipeline<D>(
```

Comment:
I like the idea of having separate methods here, though I think we should think about what we want to do with regular `aggregate` as well when deciding how to do this here for the sake of consistency. Something like `aggregate_with_type` seems like it would work well there as well, what do you think?

The main drawback in both cases is the lack of turbofish as you point out, but hopefully that won't turn out to be to much of a pain point.

Reply:
Extending `aggregate` API to offer a similar typed interface runs into the same issue that this does with sessions, see below.

Reply:
With `ChangeStream` now using `with_type`, the equivalent for `aggregate` will be `Cursor::with_type`.

> API looks good, mostly documentation suggestions. we'll also need to add the sync API, but I think it would be fine to file a separate ticket for that work

Filed RUST-1104 to track the sync API.